R/xp_utils.R

```diff
@@ -124,3 +124,14 @@ xpath_comment_re <- rex::rex(
   ":)"
 )
 xp_strip_comments <- function(xpath) rex::re_substitutes(xpath, xpath_comment_re, "", global = TRUE)
+
+#' Combine two or more nodesets to a single nodeset
+#'
+#' Useful for calling `{xml2}` functions on a combined set of nodes obtained using different XPath searches.
+#'
+#' @noRd
+combine_nodesets <- function(...) {
```

Comment:
```suggestion
TODO(r-lib/xml2#433): remove this and just use c()
combine_nodesets <- function(...) {
```

Reply:
Alternatively, we could just go ahead and rename this `c.xml_nodeset()` and register the S3 method, right?

PS FWIW it looks like package {do} defines this method doing something totally different:

https://github.com/search?q=lang%3AR%20%2Fc%5B.%5Dxml_nodeset%2F&type=code

Reply:
Not "totally". It's a by-foot version of `c()` where attributes of `..1` are kept in tact.